marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/matcher-impl/matcher-impl.xqy

```diff
@@ -361,6 +375,55 @@ declare function match-impl:build-boost-query(
   ))
 };
 
+(:
+ : Build a series of queries that match items with present but
+ : different values for declared-unique fields.
+ : @param $document  the source document from which property values are drawn
+ : @param $is-json  true if source document is JSON
+ : @param $values-by-qname  map:map that organizes document values by QName
+ : @param $match-options  XML or JSON representation of match options
+ : @return a series of cts:and-query items
+ :)
+declare function match-impl:build-uniqueness-queries(
+  $document as node(),
+  $is-json as xs:boolean,
+  $values-by-qname as map:map,
+  $match-options as item()
+)
+{
+  let $options :=
+    if ($match-options instance of object-node()) then
+      opt-impl:options-from-json($match-options)
+    else
+      $match-options
+  for $property-def in $options/matcher:property-defs/matcher:property
+    return if (xs:boolean($property-def/matcher:unique)) then
+      let $qname :=
+        fn:QName($property-def/@namespace, $property-def/@localname)
+      let $values := $values-by-qname => map:get(xdmp:key-from-QName($qname))
+      return if ($is-json) then
```

Comment:
In this section, we should adjust to make use of `helper:property-name-to-query` to make sure we pick up enhancements, like making use of range indexes, etc. An example of usage can be found here: https://github.com/marklogic/marklogic-data-hub/blob/v5.0.2/marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/algorithms/double-metaphone.xqy#L61 

Basically, passing the match options and the property name, it will return the proper cts:query for the property to values and weight to.